### PR TITLE
Create, update, delete and read Projects.

### PR DESCRIPTION
**Pull request by minuscorp**
How do we handle the ci-config repo changes? (discussion here?)

- [x] Refactor endpoints to be wherever they must be.

- [ ] Provide an API to delete existing projects (and their already done builds?).

> To make it open, we think we're going to do the following: there're two ways of deleting a project, 1) virtually, the project gets disabled (or another param) and all its build data is preserved. 2) physically, the project gets deleted from the data source and all build data and artifacts are deleted.

- [x] Provide an API to commit and push changes on the `ci-config` repo easily.

> Whose responsibility should be the operations over the `ci-config` repo? the data source? the service?

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.3.0
+2.3.3
```
Comment (KrauseFx) [inline .ruby-version]: Those changes are not actually related to this PR right? Something went wrong when rebasing?

Comment (minuscorp) [reply .ruby-version]: I rebased on master 😔 . Don't know why it included that change on my PR 🤔 . I can try to rebase again.

Comment (KrauseFx) [reply .ruby-version]: Nah, I guess it's fine... happens to me also a lot

**Comment by minuscorp** [comment]
I think I fixed the mess.

File: features/project/views/new_project.erb~Refactoring and additions [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<div class="mdl-layout mdl-js-layout mdl-layout--fixed-drawer">
```
Comment (taquitos) [inline features/project/views/new_project.erb~Refactoring and additions]: Looks like this is left over

File: features/project/views/new_project.erb~refs/remotes/fastlane/master [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<div class="mdl-layout mdl-js-layout mdl-layout--fixed-drawer">
```
Comment (taquitos) [inline features/project/views/new_project.erb~refs/remotes/fastlane/master]: This file too, extra?

File: features/project/views/new_project_form.erb~Refactoring and additions [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<div class="mdl-layout mdl-js-layout mdl-layout--fixed-drawer">
```
Comment (taquitos) [inline features/project/views/new_project_form.erb~Refactoring and additions]: And this one :)

File: features/project/views/new_project_form.erb~refs/remotes/fastlane/master [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<div class="mdl-layout mdl-js-layout mdl-layout--fixed-drawer">
```
Comment (taquitos) [inline features/project/views/new_project_form.erb~refs/remotes/fastlane/master]: Woof

File: services/config_data_sources/json_project_data_source.rb [hunk 6]
```diff
@@ -92,14 +92,14 @@ def save_git_repo_configs!(git_repo_configs: nil)
     end
 
     def create_project!(name: nil, repo_config: nil, enabled: nil, lane: nil)
-      projects = @projects
+      projects = self.projects
```
Comment (taquitos) [inline services/config_data_sources/json_project_data_source.rb]: ❤️

**Comment by taquitos** [comment]
So close 😄

**Comment by taquitos** [comment]
@minuscorp to follow up with more detailed deletion scheme.

**Pull request merged**